### ISSUE TITLE
incorrect completions on nested generic type like std.ArrayList

### ISSUE THREAD

**Issue by CGQAQ** [zigtools/zls#1392]
### Zig Version

0.12.0-dev.1+155c27f   build with -Ddata-version="0.11.0"

### Zig Language Server Version

0.12.0-dev.28+81bfc90

### Steps to Reproduce

code completion of `std.ArrayList` is completely broken

### Expected Behavior

should give me either `x` or `y` of `struct Foo`

### Actual Behavior

![image](https://github.com/zigtools/zls/assets/15936231/03ff6557-60e8-441b-a405-2713995cdcf8)
![image](https://github.com/zigtools/zls/assets/15936231/73c8ab48-33d6-422f-9496-8a2203092e71)

Sometimes will EVEN give the wrong complete items
![image](https://github.com/zigtools/zls/assets/15936231/3071ff7b-38e2-4980-a0bf-cad8534bae19)
But my `collected_data_list` is `ArrayList` of `ArrayList`
![image](https://github.com/zigtools/zls/assets/15936231/7199479d-aec4-4416-a71e-5b56a17a4327)

### P.S.
Other code completion works
![image](https://github.com/zigtools/zls/assets/15936231/c6d24b20-5e87-460c-bc98-b66905a3a397)

<details><summary>The code</summary>
<p>

```zig
const std = @import("std");

pub fn main() !void {
    const allocator = std.heap.page_allocator;

    const Foo = struct {
        x: u8,
        y: u8,
    };
    
    const ArrayList = std.ArrayList;
    const list_of_list = ArrayList(Foo).init(allocator);
    defer list_of_list.deinit();

    list_of_list.

    for (list_of_list.items) |list| {
        _ = list;
        // list.
    }
}

``` 

</p>
</details>

**Comment by nullptrdevs** [comment]
Would you happen to be using Copilot?

**Comment by CGQAQ** [comment]
> Would you happen to be using Copilot?

Yes, I am using Copilot

**Comment by Xefreh** [comment]
did you tried latest commit, code completion was not working for me before but right now it does

**Comment by CGQAQ** [comment]
> did you tried latest commit, code completion was not working for me before but right now it does

no, just checkout the latest commit same for me, and I disabled copilot as well

**Comment by nullptrdevs** [comment]
There's nothing wrong w/ Copilot, it's just that I knew ZLS doesn't provide completions for any of those but `list_of_list.`
(there's https://github.com/zigtools/zls/pull/1373 but it is quite partial and pretty sure doesn't work for any of these)

**Comment by CGQAQ** [comment]
the latest commit without copilot still gives me the wrong completion items of `list_of_list`

![image](https://github.com/zigtools/zls/assets/15936231/fb0a0031-9af4-42ac-9380-5103b32e6658)

**Comment by CGQAQ** [comment]
```zig
const std = @import("std");

const Foo = struct {
    a: i32
};
const ListOfListOfFoo = std.ArrayList(std.ArrayList(Foo));

pub fn main() !void {
    var foo_list_of_list: ListOfListOfFoo = undefined;


    for (foo_list_of_list.items) |foo_list| {
        foo_list. // gives me `a`
    }
}
```

![image](https://github.com/zigtools/zls/assets/15936231/c5ad2bcc-f883-4ab5-b56a-6a644e7893b8)

**Comment by Techatrix** [comment]
here is a reduction that doesn't use `std.ArrayList`:
```zig
fn Boxed(comptime T: type) type {
    return struct { inner: T };
}
const Some = struct { alpha: u32 };
const boxed_boxed_some: Boxed(Boxed(Some)) = undefined;
const boxed_some = boxed_boxed_some.inner;
const some = boxed_some.inner;
//    ^^^^ hovering here will show 'Boxed(...)' but should be 'Some'
```
